### DISCUSSION TITLE
`pnpm setup` perhaps should not check if PNPM_HOME is inside PATH

### DISCUSSION THREAD

**Discussion by ForgottenR** [pnpm/pnpm#15992] [Ideas]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Reproduction steps

running `pnpm setup`

### Describe the Bug

`pnpm setup` currently appends to `~/.zshrc` with code which checks if PATH already contains PNPM_HOME. But in some cases ~/.zshrc could be run multiple times, making the pnpm path not take priority (notably not over the node bin path which is installed by npm). Thus in a case where we installed pnpm via npm, and then follow the instruction to upgrade pnpm itself via pnpm, the updated version doesn't work in a vscode terminal which effectively triggers ~/.zshrc more than once (at least once when launching vscode and inherited, another when launching the terminal).

### Expected Behavior

Just don't do the checking and work like all other shell profiles normally do. Simply append.

### Which Node.js version are you using?

20.11.1

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux

**Comment by Shuuren** [comment]
I'm working on a fix for this.
